packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -2703,19 +2700,16 @@ export class AdapterClass extends EventEmitter {
                     );
                     return tools.maybeCallbackWithError(options.callback, 'common.custom needs to be an object');
                 }
-            } else {
-                if (
-                    Object.prototype.hasOwnProperty.call(options.obj.common, 'custom') &&
-                    options.obj.common.custom !== null
-                ) {
+            } else if (options.obj.common) {
+                if ('custom' in options.obj.common && options.obj.common.custom !== null) {
                     this._logger.warn(
                         `${this.namespaceLog} setObject ${options.id} (type=${options.obj.type}) property common.custom must not exist.`
                     );
                     delete options.obj.common.custom;
                 }
             }
 
-            if (!Object.prototype.hasOwnProperty.call(options.obj.common, 'name')) {
+            if (options.obj.common && !Object.prototype.hasOwnProperty.call(options.obj.common, 'name')) {
```

Comment:
Maybe
```suggestion
            if (options.obj.common && !('name' in options.obj.common)) {
```

Reply:
Unless that breaks something.

Reply:
I tried this, but then in the assignment below he says `name` does not exist on never, has common "always" has name